### ISSUE TITLE
Audio Duplication with ReactPlayer in Next.js using --turbo Flag

### ISSUE THREAD

**Issue by l-Nuril-l** [cookpete/react-player#1895]
#### Current Behavior
When using `ReactPlayer` in a Next.js project with the `--turbo` flag enabled, audio is duplicated starting from version 2.14 up to 2.16 of `react-player`. The issue does not occur in earlier versions (e.g., `2.13`). This happens when toggling the `ReactPlayer` component based on state.

#### Expected Behavior
Audio should play without duplication when `ReactPlayer` is triggered, and the sound should play as expected without overlapping.

#### Steps to Reproduce
1. Use Next.js with the `--turbo` flag (`"dev": "next dev --turbo"`).
2. Include the `ReactPlayer` component that streams video from a local or remote source.
3. Toggle the player using a button to initiate playback.
4. Observe that, starting from `react-player` versions 14-16, the sound is duplicated.

#### Environment
- URL attempting to play: `http://localhost:5284/api/video/stream/2`
- Browser: Chrome 129
- Operating system: Windows 11

#### Code Example
```js
"use client"
import React, { useState } from 'react';
import ReactPlayer from 'react-player';

export default function page() {
    const [state, setState] = useState(false);
    return (
        <div>
            <button onClick={() => setState(true)}>click</button>
            {state && <ReactPlayer url='https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4' playing={true} />}
        </div>
    );
}

**Comment by Deamoner** [comment]
I have this same issue. Any update?

**Comment by Deamoner** [comment]
Pretty sure this has to do with react strict mode in nextjs. In prod doesn't happen and seems to be temporary fixable with nextjs config change:
module.exports = {
  // Only do this for testing - don't commit this change
  reactStrictMode: false,
  // ... other config
}

**Comment by arjuna-wenzel** [comment]
I'm also having the same issue, but for me it's both in an built app and in dev

**Comment by JoseRaulLH** [comment]
same issue here

**Comment by vincentwinkel** [comment]
Same issue here.

The issue occurs only for media files (no embedded players) and only if they are the 1st media to be played (eg. I play first youtube then mp3, no problem, then I reload the page and play directly mp3 -> audio duplicated).

**Comment by josephmalicke** [comment]
Yes, this is a major issue for me. I confirm that the there is a duplicate audio issue that exists in NextJs for external media files and ONLY if it is the 1st media to be played. If it is the 1st media to be played, and if it is played after initial load, with playing=true, then there is a duplicate audio stream.

**Comment by luwes** [comment]
Is this still an issue in v3? please provide a test example for debugging purposes, thanks!

**Comment by vincentwinkel** [comment]
Personally I wont move to v3. You dropped half of the platforms and I'm not sure to understand how this project is still useful, except promoting Mux.

**Comment by luwes** [comment]
I see how you might interpret it this way. The main reason we dropped those platforms are because there are no [media-elements](https://github.com/muxinc/media-elements) available yet for those platforms. We decided to use our media-elements in ReactPlayer to not have to duplicate work and to support custom controls via Media Chrome. 

We're happy to add them back if a media element is created for them. 
It's not with bad intention we removed those. Someone has to maintain all these things.

See https://github.com/cookpete/react-player/discussions/1843

**Comment by Aluisio** [comment]
I use this lib in my SaaS basically for the progressInterval and onProgress properties to save the progress watched by the user at each time period. However, this was removed in v3, and for me to implement this logic manually is the same work as using an iFrame, listening to the events and doing the processing (as I have already done for other players that the lib does not support). 
I will probably also stop using this lib

**Comment by luwes** [comment]
@Aluisio thanks for the feedback. Could you please add that feature request to https://github.com/cookpete/react-player/discussions/categories/ideas so we can gauge interest. If there are enough votes it can be added back in. 

The idea with v3 was also to start with a clean slate and make the API as simple and as inline with the HTMLVideoElement API as possible.

**Comment by l-Nuril-l** [comment]
This issue is no longer present in v3.

**Comment by jasoncalk** [comment]
Also hit this on react-player@2.16.1 in a Next.js 15 App Router app (Server Components + client MediaPlayer wrapper), and wanted to add a data point since a couple of things in this thread matched our repro exactly.

Trigger: Matches what @vincentwinkel and @robertmalicke described — it's specifically the first real (non-preview) player activation on a given page load. Any player works fine as long as something else has already played first; a lone player, first to activate, reproduces it reliably. I built a minimal isolated test (a single bare <ReactPlayer> in a popup, nothing else on the page) specifically to rule out anything component-specific on our end, and it reproduced the double audio immediately.

Prod vs. dev: Also confirming @arjuna-wenzel over @Deamoner's Strict Mode theory — I reproduced this in a full production build (next build && next start, no dev server, no Strict Mode) as well as in dev. Disabling reactStrictMode didn't fix it for us.

Diagnosis: chrome://media-internals showed two separate WebMediaPlayer instances created ~200–300ms apart for a single, stable DOM element/React mount — so it's a duplicate native player backing one player instance, not a duplicate React element on our side.

Workaround (v2, until we can evaluate a v3 migration): I rendered one hidden, silent, always-paused ReactPlayer once globally on every page load (in our root layout/theme provider), pointed at a URL that doesn't resolve to real content, with preload: 'none'. This forces whatever lazy-loaded player chunk/module resolution is racing to happen once, harmlessly, before any real user-triggered playback — since then I haven't been able to reproduce it the issue. 

Leaving this here in case it's useful to anyone else still on v2 who lands on this closed issue.

Haven't had a chance to verify against v3 firsthand yet (API migration is a bigger lift for us), so can't confirm/deny that part — just wanted to corroborate the v2 root cause and leave a workaround for others in the meantime.
